include/boost/geometry/extensions/random/strategies/agnostic/uniform_linear.hpp

```diff
@@ -0,0 +1,201 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2019 Tinko Bartels, Berlin, Germany.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_EXTENSIONS_RANDOM_STRATEGIES_AGNOSTIC_UNIFORM_LINEAR_HPP
+#define BOOST_GEOMETRY_EXTENSIONS_RANDOM_STRATEGIES_AGNOSTIC_UNIFORM_LINEAR_HPP
+
+#include <random>
+#include <iterator>
+#include <algorithm>
+
+#include <boost/geometry/algorithms/equals.hpp>
+#include <boost/geometry/core/point_type.hpp>
+#include <boost/geometry/core/cs.hpp>
+#include <boost/geometry/algorithms/length.hpp>
+#include <boost/geometry/algorithms/line_interpolate.hpp>
+#include <boost/geometry/strategies/cartesian/line_interpolate.hpp>
+#include <boost/geometry/strategies/geographic/line_interpolate.hpp>
+#include <boost/geometry/strategies/spherical/line_interpolate.hpp>
+
+#include <boost/geometry/extensions/random/strategies/uniform_point_distribution.hpp>
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace uniform_point_distribution {
+
+template
+<
+    typename Point,
+    typename DomainGeometry,
+    typename InterpolationStrategy =
+        typename strategy::line_interpolate::services::default_strategy
+            <
+                typename cs_tag<DomainGeometry>::type
+            >::type
+>
+struct uniform_linear_single
+{
+    uniform_linear_single(DomainGeometry const& g) {}
+    bool equals(DomainGeometry const& l_domain,
+                DomainGeometry const& r_domain,
+                uniform_linear_single const& r_strategy) const
+    {
+        return boost::geometry::equals(l_domain, r_domain);
+    }
+    
+    template<typename Gen>
+    Point apply(Gen& g, DomainGeometry const& d)
+    {
+        typedef typename select_most_precise
+            <
+                double,
+                typename coordinate_type<Point>::type
+            >::type sample_type;
+        std::uniform_real_distribution<sample_type> dist(0, length(d));
+        sample_type r = dist(g);
+        Point p;
+        boost::geometry::line_interpolate(d, r, p);
```

Comment:
`line_interpolate()` needs a strategy as well.

Would it make sense to make `uniform_linear_single` a part of the randomization algorithm (so algorithm dispatched for single linear geometries) and add umbrella strategy (or 3, one for each CS) containing getters for `distance`/`length` and `line_interpolate` strategies?

It's also the case for multi strategy below and in fact AFAIU in both cases (single and multi) the same umbrella strategy could be used because below also `length` and `line_interpolate` is called.